sql/core/src/main/scala/org/apache/spark/sql/RuntimeConfig.scala

```diff
@@ -154,5 +154,9 @@ class RuntimeConfig private[sql](sqlConf: SQLConf = new SQLConf) {
     if (SQLConf.staticConfKeys.contains(key)) {
       throw new AnalysisException(s"Cannot modify the value of a static config: $key")
     }
+    if (sqlConf.setCommandRejectsSparkConfs &&
+        ConfigEntry.findEntry(key) != null && !SQLConf.sqlConfEntries.containsKey(key)) {
```

Comment:
Actually, thinking of it, isn't `ConfigEntry.findEntry(key) != null` redundant with the other check?

Removing that would also want by other settings that don't have constants defined.

Reply:
I'm sorry for the delay.
As per the comment https://github.com/apache/spark/pull/22887#issuecomment-442425557, I'd leave it as is for now.

Reply:
we should only reject configs that are registered as SparkConf. Thinking about configs that are either a SparkConf or SQLConf, we shouldn't reject it.